csep/utils/readers.py

```diff
@@ -473,8 +473,69 @@ def parse_datetime(dt_string):
         else:
             return events, catalog_id
 
-def read_em_gcmt(fname):
-    raise NotImplementedError("not implemented yet!")
+
+
+def read_ingv_rcmt_csv(fname):
+    """
+    Reader for the INGV (Istituto Nazionale di Geofisica e Vulcanologia - Italy)  European-
+    Mediterranean regional Centroid Moment Tensor Catalog.
+    It reads a catalog in .csv format, directly downloaded from http://rcmt2.bo.ingv.it/ using the Catalog Search (Beta
+    version).
+
+    The ZMAP Format has the following dtype:
+
+    dtype = numpy.dtype([('longitude', numpy.float32),
+                        ('latitude', numpy.float32),
+                        ('year', numpy.int32),
+                        ('month', numpy.int32),
+                        ('day', numpy.int32),
+                        ('magnitude', numpy.float32),
+                        ('depth', numpy.float32),
+                        ('hour', numpy.int32),
+                        ('minute', numpy.int32),
+                        ('second', numpy.int32)])
+    """
+
+    import time             ### Should these modules be imported in top level of readers.py?
+    import pandas
+
+    start_time = time.time()
+
+    indexing = {'date': 1,
+                'time': 2,
+                'sec_dec': 3,
+                'hypo_lat': 4,
+                'hypo_lng': 5,
+                'hypo_depth_in_km': 6,
+                'Mw': 61}
+
+    data = pandas.read_csv(fname, header=None, usecols=list(indexing.values()))
```

Comment:
seems ok, but would we rather use Numpy dependency here for reader?

Reply:
There would not be a problem, for consistency. But I thought that pandas or even csv modules, are much faster to read big files than numpy (and memory saving). Especially, in these catalogs that have a lot of information that we wouldn't need.   Let me know what you think.

Reply:
im actually not sure about the performance difference. it might be worth doing a test to confirm this. for the csep-csv format i've used the python csv module to handle the reading.

Reply:
Lets use csv then for consistency! Sounds better. Ill modify it tomorrow c:

Reply:
Lets use csv then for consistency! Sounds better. Ill modify it tomorrow c:

Reply:
cool sounds good. (: since you have the pandas implemented already can you check the timing using the timeit module?

Reply:
forgot to say that the csv module is 10x times faster than pandas. ooof